pkg/utils/functional/functional.go

```diff
@@ -152,6 +152,16 @@ func ValidateAll(errorables ...func() error) error {
 	return err
 }
 
+// All returns true if all the predicate funcs return true and false otherwise.
+func All(predicates ...func() bool) bool {
```

Comment:
AllTrue?

Reply:
If we go with this route:

```
   if isInstanceTypeSupported(constraints.InstanceTypes, instanceTypeInfo) && 
      isArchitectureSupported(architecture, instanceTypeInfo) &&
      isCapacityTypeSupported(constraints.CapacityType, instanceTypeInfo) &&
      isZonesSupported(zones, instanceTypeInfo) {
     ...
     }
```

I'll just delete this func.  Or we can change the signature to `func AllTrue(bools ...bool) bool`

Reply:
prefer the former.